### PR TITLE
Add tests for the appropriate download source (RhBug:1775184)

### PR DESCRIPTION
**Pull request by kontura**
For:
https://github.com/rpm-software-management/librepo/pull/184
https://github.com/rpm-software-management/libdnf/pull/911

https://bugzilla.redhat.com/show_bug.cgi?id=1775184

### CODE REVIEW

**Comment by kontura** [comment]
I have added @not.with_os=rhel__eq__8.

**Comment by kontura** [comment]
I have added a few additional test cases that require: https://github.com/rpm-software-management/librepo/pull/189

**Comment by lukash** [comment]
@Conan-Kudo the librepo PR you've merged: https://github.com/rpm-software-management/librepo/pull/189 comes with these tests here. To maintain a sane workflow, we need the tests to be reviewed and merged along with the implementation. You should also make sure the tests cover the implementation well and that they actually work. I know this makes merging PRs a lot more work, but I hope you understand we need to do it this way... If you need something from us to be able to do this, let us know. Thanks!

**Comment by Conan-Kudo** [comment]
@lukash how do I even run these? And I _still_ don't have access to view your CI system...

**Comment by lukash** [comment]
> @lukash how do I even run these?

That's the question, right? Not straightforward. @dmach has this tool here: https://github.com/dmach/dbox/ That is meant to make it easy to run stuff against development builds of the dnf stack (without having to build packages). It works for the most part (just follow the usage, it's only missing running the stack itself :sunglasses: ), though there was a weird issue with gpg agent generating the test data in the CI stack. I'd say that's just if you want to dive in yourself though and we should prepare some more complete howto.

> And I _still_ don't have access to view your CI system...

True... But it's not relevant if you run it yourself...

File: dnf-behave-tests/features/download-source.feature [hunk 1]
```diff
@@ -0,0 +1,93 @@
+Feature: Tests for picking the appropriate download source
+
+
+@not.with_os=rhel__eq__8
+@bz1775184
+Scenario: baseurl is used if all mirrors from mirrorlist fail
+Given I create directory "/baseurlrepo"
+  And I execute "createrepo_c /{context.dnf.installroot}/baseurlrepo"
```
Comment (lukash) [inline dnf-behave-tests/features/download-source.feature:8]: Nitpick: the `/` in `/{context.dnf.installroot}/baseurlrepo` is superfluous. (in all scenarios)

Comment (kontura) [reply dnf-behave-tests/features/download-source.feature]: :+1:

**Pull request merged**